src/psm/src/gmat.cpp

```diff
@@ -287,25 +284,52 @@ void GMat::GenerateStripeConductance(int t_l,
     NodeMap::iterator x_prev;
     // int               y_loc = (t_y_min + t_y_max) / 2;
     int i = 0;
-    for (x_itr = layer_map.lower_bound(t_x_min);
-         x_itr != layer_map.upper_bound(t_x_max); ++x_itr) {
-      map<int, Node*>::iterator y_itr = (x_itr->second).lower_bound(t_y_min);
+    NodeMap::iterator x_start = layer_map.lower_bound(t_x_min);
```

Comment:
This is really convoluted and hard to follow.  What's going on here and can it be refactored / clarified?